src/bosh-director/lib/bosh/blobstore_client/azurestoragecli_blobstore_client.rb

```diff
@@ -0,0 +1,138 @@
+require 'openssl'
+require 'digest/sha1'
+require 'base64'
+require 'securerandom'
+require 'open3'
+require 'json'
+
+module Bosh::Blobstore
+  class AzurestoragecliBlobstoreClient < BaseClient
+    # Blobstore client for azure storage account, using azure-storage-cli Go version
+    # @param [Hash] options azure storage account connection options
+    # @option options [Symbol] account_name
+    #   key that is applied before the object is sent to azure storage account
+    # @option options [Symbol, optional] account_key
+    # @option options [Symbol] azure_storage_cli_path
+    #   path to azure-storage-cli binary
+    # @option options [Symbol, optional] azure_storage_cli_config_path
+    #   path to store configuration files
+    def initialize(options)
+      super(options)
+
+      @azure_storage_cli_path = @options.fetch(:azure_storage_cli_path)
+
+      unless Kernel.system(@azure_storage_cli_path.to_s, '--v', out: '/dev/null', err: '/dev/null')
+        raise BlobstoreError, 'Cannot find azure-storage-cli executable. Please specify azure_storage_cli_path parameter'
+      end
+
+      @azure_storage_cli_options = {
+        "account-name": @options[:account_name],
+        "container-name": @options[:container_name],
+        "account-key": @options[:account_key]
+      }
+
+      @azure_storage_cli_options.reject! { |_k, v| v.nil? }
+
+      @config_file = write_config_file(@azure_storage_cli_options, @options.fetch(:azure_storage_cli_config_path, nil))
+    end
+
+    def redacted_credential_properties_list
+      %w[account_key]
+    end
+
+    def encryption_headers; end
+
+    def encryption?
+      false
+    end
+
+    protected
+
+    # @param [File] file file to store in az storage account
+    def create_file(object_id, file)
+      object_id ||= generate_object_id
+      # in Ruby 1.8 File doesn't respond to :path
+      path = file.respond_to?(:path) ? file.path : file
```

Comment:
we are not using ruby 1.8 since a long while. Do we really need that check here?

Reply:
For this I'm not sure whether ruby 1.8 is never used. Could someone please confirm that?

Reply:
Can be removed we do not have to support ruby 1.8

Reply:
Done